naive-serverless-map-reduce/mr.js

```diff
@@ -0,0 +1,58 @@
+const request = (() => {
+  const https = require('https');
+  const agent = new https.Agent({ keepAlive: true });
+  return (method, host, path, headers, body) => {
+    return new Promise((resolve, reject) => {
+      const opts = { method, host, path, headers, agent, port: 443 };
+      const req = https.request(opts, res => {
+        const chunks = [];
+        res.on('data', chunk => chunks.push(chunk));
+        res.on('end', () => (res.statusCode === 200 ? resolve : reject)({
+          status: res.statusCode,
+          headers: res.headers,
+          body: chunks.join('')
+        }));
+      });
+      req.on('error', e => reject({
+       status: -1,
+        headers: {},
+        body: `Network error: ${e}`
+      }));
+      req.write(body);
+      req.end();
+    });
+  };
+})();
+
+const invoke = async (fname, fargs) => {
+  console.log(`Invoking ${fname}()`);
+  if (typeof(fname) !== 'string' || !/^[a-zA-Z_][a-zA-Z0-9_]*$/.test(fname)) {
```

Comment:
Validating with the regex isn't helpful. I could break this code if I wanted so client side validation isn't worth it.